tests/small_model_debugging/test_ulysses_hf.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+
+import deepspeed.comm as dist
+import deepspeed
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from torch.utils.data import DataLoader
+import numpy as np
+import transformers
+from deepspeed.sequence.cross_entropy import vocab_sequence_parallel_cross_entropy
+
+MODEL_NAME = "facebook/opt-125m"
+
+DS_CONFIG = {
+    'bf16': {
+        'enabled': True
+    },
+    'optimizer': {
+        'type': 'AdamW',
+        'params': {
+            'lr': 2e-05,
+            'betas': [0.9, 0.999],
+            'eps': 1e-08,
+            'weight_decay': 0.0
+        }
+    },
+    'scheduler': {
+        'type': 'WarmupLR',
+        'params': {
+            'warmup_min_lr': 0,
+            'warmup_max_lr': 2e-05,
+            'warmup_num_steps': 0
+        }
+    },
+    'zero_optimization': {
+        'stage': 1,
+        'overlap_comm': True,
+        'contiguous_gradients': True,
+        'sub_group_size': 1e9,
+        'sub_group_size': 1000000000.0,
+        'reduce_bucket_size': 16777216,
+        'stage3_prefetch_bucket_size': 15099494.4,
+        'stage3_max_live_parameters': 1000000000.0,
+        'stage3_max_reuse_distance': 1000000000.0,
+        'stage3_gather_16bit_weights_on_model_save': True
+    },
+    'gradient_accumulation_steps': 1,
+    'gradient_clipping': 1.0,
+    'steps_per_print': float('inf'),
+    'train_batch_size': 1,
+    'train_micro_batch_size_per_gpu': 1,
+    'wall_clock_breakdown': False,
+}
+
+device = "cuda:0" if torch.cuda.is_available() else "cpu"
+
+
+def load_and_prepare_data(model_name):
+    """Load model, tokenizer and dataset, and prepare data loader."""
+    from datasets import load_dataset
+    # Load model and tokenizer
+    config = transformers.AutoConfig.from_pretrained(model_name, trust_remote_code=True)
+    config.num_hidden_layers = 9
+    model = AutoModelForCausalLM.from_pretrained(model_name,
+                                                 config=config,
+                                                 trust_remote_code=True,
+                                                 torch_dtype=config.torch_dtype,
+                                                 attn_implementation="flash_attention_2")
+
+    model.to(device)
+
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
+    tokenizer.pad_token = tokenizer.eos_token
+
+    # Load and tokenize dataset
+    dataset = load_dataset("wikitext", 'wikitext-103-raw-v1', split='train[:1%]').filter(lambda x: x['text'])
```

Comment:
Downloading a dataset for a unit test can be a load for the server and network and has a risk that the access is blocked. Can we use the synthetic data for this?
The same goes for models. You can use the HF model definition to create a model and random weights instead of downloading the trained weights.

Reply:
This is more of an integration than unit test, but I see your point, will replace with synthetic unit test(s) and move this to/augment existing DeepSpeedExamples example.